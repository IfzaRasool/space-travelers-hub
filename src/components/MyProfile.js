/* eslint-disable no-console */
import React from 'react';
import './MyProfile.css';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missionStore = useSelector((state) => (state.missionReducer));
  console.log(missionStore);

  return (
    <div>
      <div className="myProfile-container">
        <div className="mission-item">
          <h3> My Missions</h3>
          <h6>dfds</h6>
        </div>
        <div>
          <h3>My Rockets</h3>
        </div>
      </div>
    </div>
  );
};
// const MyProfile = () => {
//   const missionStore = useSelector((state) => state.missionReducer);

//   console.log(missionStore);
//   return (
//     <div>
//       <div className="myProfile-container">
//         <div className="mission-item">
//           <h3> My Missions</h3>
//           {/* { missionStore.map((e) => (
//             <h6 key={e.id}>
//               {e.name}
//             </h6>
//           ))} */}
//         </div>
//       </div>
//       <div>
//         <h3>My Rockets</h3>
//       </div>
//     </div>
//   );
// };

export default MyProfile;
