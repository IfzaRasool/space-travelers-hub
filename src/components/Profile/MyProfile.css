@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap');

* {
  font-family: 'Poppins', sans-serif;
}

.myProfile-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  padding: 2rem;
  gap: 50px;
}

.reserved-container {
  border: 1px solid rgb(10, 9, 9);
  display: flex;
  flex-direction: column;
  border-radius: 7px;
  overflow: hidden;
}

.rocket-name {
  border: 1px solid rgb(10, 9, 9);
  padding: 0.5rem;
}

.rocket-name:nth-child(odd) {
  border-top: 2px rgb(185, 183, 183) solid;
}
