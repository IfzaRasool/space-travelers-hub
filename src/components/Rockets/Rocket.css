@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap");

.rocket-container {
  padding: 10px 8%;
  display: flex;
  flex-direction: column;
  gap: 30px;
  font-family: "Poppins", sans-serif;
}

.rocket-card {
  display: flex;
  gap: 20px;
}

.img-box {
  width: 30%;
  max-height: 100%;
}

img {
  height: 100%;
  width: 100%;
}

.text-box {
  width: 70%;
  display: flex;
  flex-direction: column;
  gap: 8px;
  align-items: flex-start;
}

.text-box > button {
  padding: 10px 15px;
  background-color: rgb(80, 80, 247);
  color: rgb(240, 240, 240);
  border: none;
  border-radius: 4px;
  font-size: 0.91rem;
  cursor: pointer;
  font-family: "Poppins", sans-serif;
}

.text-box > h2 {
  color: rgb(26, 26, 26);
}

.reserved-badge{
  background-color: darkcyan;
  border-radius: 3px;
  color: #fff;
  padding: 2px 3px;
}

@media screen and (max-width:992px){
  .text-box{
    font-size: 13px;
    width:60%;
  }

   .img-box{
     width: 40%;
   }

   .text-box>button{
     font-size: 0.75rem;
   }
}