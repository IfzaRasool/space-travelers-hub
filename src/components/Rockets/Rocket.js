import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Rocket = (props) => {
  const [reserved, setReserved] = useState(false);
  const {
    name, description, flickrImage, handleClickProp, id,
  } = props;
  return (
    <article className="rocket-card">
      <div className="img-box">
        <img src={flickrImage} alt="rocket" />
      </div>
      <div className="text-box">
        <h2>{name}</h2>
        <p>{description}</p>
        <button
          onClick={() => {
            setReserved(!reserved);
            handleClickProp(id);
          }}
          type="button"
        >
          {reserved ? 'Cancel Reservation' : 'Reserve Rocket'}

        </button>
      </div>
    </article>
  );
};

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickrImage: PropTypes.string.isRequired,
  handleClickProp: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
};

export default Rocket;
