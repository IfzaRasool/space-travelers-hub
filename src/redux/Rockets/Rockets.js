const GET_ROCKETS = 'GET_ROCKETS';
const BOOK_ROCKET = 'BOOK_ROCKET';
const CANCEL_ROCKET = 'CANCEL_ROCKET';
const BASE_URL = 'https://api.spacexdata.com/v3';
const END_POINT = 'rockets';

const Reducer = (state = [], action) => {
  switch (action.type) {
    case GET_ROCKETS:
      return action.rockets.map((rocket) => {
        const {
          id,
          rocket_name: rocketName,
          description,
          flickr_images: flickrImages,
        } = rocket;
        return {
          id,
          rocketName,
          description,
          flickrImages,
        };
      });
    case BOOK_ROCKET:
      return state.map((rocket) => {
        if (rocket.id === action.id) {
          return { ...rocket, reserved: true };
        }
        return rocket;
      });
    case CANCEL_ROCKET:
      return state.map((rocket) => {
        if (rocket.id === action.id) {
          return { ...rocket, reserved: false };
        }
        return rocket;
      });
    default:
      return state;
  }
};

const getRockets = (rockets) => ({ type: GET_ROCKETS, rockets });

export const bookRocket = (id) => ({ type: BOOK_ROCKET, id });

export const cancelBookRocket = (id) => ({ type: CANCEL_ROCKET, id });

export const getAsyncRockets = () => (dispatch) => {
  fetch(`${BASE_URL}/${END_POINT}`, {
    headers: {
      'Content-type': 'application/json; charset=UTF-8',
    },
  })
    .then((res) => res.json())
    .then((data) => {
      dispatch(getRockets(data));
    });
};

export default Reducer;
