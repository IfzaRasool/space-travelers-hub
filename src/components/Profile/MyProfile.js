/* eslint-disable no-console */
import React from 'react';
import './MyProfile.css';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missionStore = useSelector((state) => state.missionReducer);

  const missionName = () => {
    console.log('profile');
    // const result = missionStore.filter((name) => (name.id));
    // console.log(result);
  };

  return (
    <div>
      <div className="myProfile-container">
        <div className="mission-item">
          <h3> My Missions</h3>
          <h6 onLoad={missionName()}>
            {missionStore.map((e) => e.name)}
          </h6>
        </div>
        <div>
          <h3>My Rockets</h3>
        </div>
      </div>
    </div>
  );
};

export default MyProfile;
