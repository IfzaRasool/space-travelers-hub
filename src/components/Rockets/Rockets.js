import { useDispatch, useSelector } from 'react-redux';
import Rocket from './Rocket';
import { bookRocket, cancelBookRocket } from '../../redux/Rockets/Rockets';
import './Rocket.css';

const Rockets = () => {
  const myRockets = useSelector((state) => state.Rockets);
  const dispatch = useDispatch();
  const reserveRocket = (id) => {
    dispatch(bookRocket(id));
  };
  const cancelReservation = (id) => {
    dispatch(cancelBookRocket(id));
  };
  return (
    <div className="rocket-container">
      {myRockets.map((rocket) => {
        const {
          id, rocketName, description, flickrImages, reserved,
        } = rocket;
        return (
          <Rocket
            key={id}
            name={rocketName}
            description={description}
            flickrImage={flickrImages[0]}
            handleClickProp={reserveRocket}
            handleCancelReservation={cancelReservation}
            id={id}
            reserved={reserved}
          />
        );
      })}
    </div>
  );
};

export default Rockets;
