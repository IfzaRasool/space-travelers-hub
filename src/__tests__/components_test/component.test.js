import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import MyProfile from '../../components/Profile/MyProfile';
import Rockets from '../../components/Rockets/Rockets';
import Rocket from '../../components/Rockets/Rocket';

describe('test snapshots for all components', () => {
  const myRockets = [];
  it('renders rocket page correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets>
            {myRockets.forEach(() => renderer.create(<Rocket />).toJSON())}
          </Rockets>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Profile correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders single rocket correctly', () => {
    const tree = renderer.create(<Rocket />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
