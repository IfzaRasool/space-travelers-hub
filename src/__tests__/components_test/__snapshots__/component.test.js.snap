// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshots for all components renders Profile correctly 1`] = `
<div>
  <div
    className="myProfile-container"
  >
    <div
      className="mission-item"
    >
      <h3>
         My Missions
      </h3>
      <h6 />
    </div>
    <div>
      <h3>
        My Rockets
      </h3>
      <div
        className="reserved-container"
      />
    </div>
  </div>
</div>
`;

exports[`test snapshots for all components renders rocket page correctly 1`] = `
<div
  className="rocket-container"
/>
`;

exports[`test snapshots for all components renders single rocket correctly 1`] = `
<article
  className="rocket-card"
>
  <div
    className="img-box"
  >
    <img
      alt="rocket"
    />
  </div>
  <div
    className="text-box"
  >
    <h2 />
    <p>
       
    </p>
    <button
      onClick={[Function]}
      style={
        Object {
          "transition": "all 0.3s",
        }
      }
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</article>
`;
